search_commands.go

```diff
@@ -0,0 +1,1215 @@
+	FTAlter(ctx context.Context, index string, skipInitalScan bool, definition []interface{}) *StatusCmd
+	FTConfigGet(ctx context.Context, option string) *MapStringInterfaceCmd
+	FTConfigSet(ctx context.Context, option string, value interface{}) *StatusCmd
+	FTCreate(ctx context.Context, index string, options *FTCreateOptions, schema ...*SearchSchema) *StatusCmd
+	FTCursorDel(ctx context.Context, index string, cursorId int) *StatusCmd
+	FTCursorRead(ctx context.Context, index string, cursorId int, count int) *MapStringInterfaceCmd
+	FTDictAdd(ctx context.Context, dict string, term []interface{}) *IntCmd
+	FTDictDel(ctx context.Context, dict string, term []interface{}) *IntCmd
+	FTDictDump(ctx context.Context, dict string) *StringSliceCmd
+	FTDropIndex(ctx context.Context, index string) *StatusCmd
+	FTDropIndexWithArgs(ctx context.Context, index string, options *FTDropIndexOptions) *StatusCmd
+	FTExplain(ctx context.Context, index string, query string) *StringCmd
+	FTExplainWithArgs(ctx context.Context, index string, query string, options *FTExplainOptions) *StringCmd
+	FTInfo(ctx context.Context, index string) *MapStringInterfaceCmd
+	FTProfileSearch(ctx context.Context, index string, limited bool, query SearchQuery) *MapStringInterfaceCmd
+	FTProfileAggregate(ctx context.Context, index string, limited bool, query AggregateQuery) *MapStringInterfaceCmd
+	FTSpellCheck(ctx context.Context, index string, query string) *MapStringInterfaceCmd
+	FTSpellCheckWithArgs(ctx context.Context, index string, query string, options *FTSpellCheckOptions) *MapStringInterfaceCmd
+	FTSearch(ctx context.Context, index string, query string) *Cmd
+	FTSearchWithArgs(ctx context.Context, index string, query string, options *FTSearchOptions) *Cmd
+	FTSynDump(ctx context.Context, index string) *MapStringSliceInterfaceCmd
+	FTSynUpdate(ctx context.Context, index string, synGroupId interface{}, terms []interface{}) *StatusCmd
+	FTSynUpdateWithArgs(ctx context.Context, index string, synGroupId interface{}, options *FTSynUpdateOptions, terms []interface{}) *StatusCmd
+	FTTagVals(ctx context.Context, index string, field string) *StringSliceCmd
+}
+
+type FTCreateOptions struct {
+	OnHash          bool
+	OnJSON          bool
+	Prefix          []interface{}
+	Filter          string
+	DefaultLanguage string
+	LanguageField   string
+	Score           float64
+	ScoreField      string
+	PayloadField    string
+	MaxTextFields   int
+	NoOffsets       bool
+	Temporary       int
+	NoHL            bool
+	NoFields        bool
+	NoFreqs         bool
+	StopWords       []interface{}
+	SkipInitalScan  bool
+}
+
+type SearchSchema struct {
+	FieldName         string
+	As                string
+	FieldType         string
+	Sortable          bool
+	UNF               bool
+	NoStem            bool
+	NoIndex           bool
+	PhoneticMatcher   string
+	Weight            float64
+	Seperator         string
+	CaseSensitive     bool
+	WithSuffixtrie    bool
+	VectorArgs        *FTVectorArgs
```

Comment:
Not all SerachSchemas require a vector. IMHO we should revisit this collection. Maybe we have a schema per type of schemafield?

Reply:
This is a low-level implementation of RedisSearch. We plan to add a high-level implementation for vector and other types in the future.